core/storage/learner_group/gae_models.py

```diff
@@ -0,0 +1,247 @@
+        """Model contains user data to be exported."""
+        return dict(super(cls, cls).get_export_policy(), **{
+            'title': base_models.EXPORT_POLICY.EXPORTED,
+            'description': base_models.EXPORT_POLICY.EXPORTED,
+            'facilitator_user_ids': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+            'student_user_ids': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+            'invited_user_ids': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+            'subtopic_page_ids': base_models.EXPORT_POLICY.EXPORTED,
+            'story_ids': base_models.EXPORT_POLICY.EXPORTED
+        })
+
+    # We have ignored [override] here because the signature of this method
+    # doesn't match with signature of super class's get_new_id() method.
+    @classmethod
+    def get_new_id(cls) -> str:  # type: ignore[override]
+        """Generates an ID for a new LearnerGroupModel.
+
+        Returns:
+            str. The new ID.
+
+        Raises:
+            Exception. An ID cannot be generated within a reasonable number
+                of attempts.
+        """
+        for _ in range(base_models.MAX_RETRIES):
+            group_id = ''.join(
+                random.choice(string.ascii_lowercase + string.ascii_uppercase)
+                for _ in range(base_models.ID_LENGTH))
+            if not cls.get_by_id(group_id):
+                return group_id
+
+        raise Exception('New id generator is producing too many collisions.')
+
+    @classmethod
+    def create(
+        cls,
+        group_id: str,
+        title: str,
+        description: str
+    ) -> LearnerGroupModel:
+        """Creates a new LearnerGroupModel instance and returns it.
+
+        Args:
+            group_id: str. The ID of the learner group.
+            title: str. The title of the learner group.
+            description: str. The description of the learner group.
+
+        Returns:
+            LearnerGroupModel. The newly created LearnerGroupModel instance.
+
+        Raises:
+            Exception. A learner group with the given group ID exists already.
+        """
+        if cls.get_by_id(group_id):
+            raise Exception(
+                'A learner group with the given group ID exists already.')
+
+        entity = cls(id=group_id, title=title, description=description)
+
+        entity.update_timestamps()
+        entity.put()
+        return entity
+
+    @classmethod
+    def export_data(cls, user_id: str) -> Dict[str, Dict[str, List[str]]]:
+        """Takeout: Export LearnerGroupModel user-based properties.
+
+        Args:
+            user_id: str. The user_id denotes which user's data to extract.
+
+        Returns:
+            dict. A dict containing the user-relevant properties of
+            LearnerGroupModel.
+        """
+        found_models = cls.get_all().filter(
+            datastore_services.any_of(
+                cls.student_user_ids == user_id,
+                cls.invited_user_ids == user_id,
```

Comment:
For invited_user_ids, such users aren't actually part of the group yet (and would never have seen the materials that make up the group), so it feels like the representation here might actually confuse a user who does takeout and sees groups that they didn't think they belonged to. Could we represent those "invited to" groups separately somehow and only show their title/description?